spec/services/providers/copy_to_recruitment_cycle_service_spec.rb

```diff
@@ -0,0 +1,122 @@
+    let(:new_provider) do
+      new_recruitment_cycle.reload.providers.find_by(
+        provider_code: provider.provider_code
+      )
+    end
+    let(:mocked_copy_course_service) { double(execute: nil) }
+    let(:mocked_copy_site_service) { double(execute: nil) }
+    let(:service) do
+      described_class.new(
+        provider: provider,
+        copy_course_to_provider_service: mocked_copy_course_service,
+        copy_site_to_provider_service: mocked_copy_site_service
+      )
+    end
+
+    it 'makes a copy of the provider in the new recruitment cycle' do
+      expect(
+        new_recruitment_cycle.providers.find_by(
+          provider_code: provider.provider_code
+        )
+      ).to be_nil
+
+      service.execute(new_recruitment_cycle)
+
+      expect(new_provider).not_to be_nil
+      expect(new_provider).not_to eq provider
+    end
+
+    it 'leaves the existing provider alone' do
+      service.execute(new_recruitment_cycle)
+
+      expect(recruitment_cycle.reload.providers).to eq [provider]
+    end
+
+    context 'the provider already exists in the new recruitment cycle' do
+      let(:new_provider) {
+        build :provider, provider_code: provider.provider_code
+      }
+      let(:new_recruitment_cycle) {
+        create :recruitment_cycle, :next,
+               providers: [new_provider]
+      }
+
+      it 'does not make a copy of the provider' do
+        expect { service.execute(new_recruitment_cycle) }
+          .not_to(change { new_recruitment_cycle.reload.providers.count })
+      end
+    end
+
+    it 'assigns the new provider to organisation' do
+      service.execute(new_recruitment_cycle)
+
+      expect(new_provider.organisation).to eq provider.organisation
+    end
+
+    it 'copies over the ucas_preferences' do
+      service.execute(new_recruitment_cycle)
+
+      compare_attrs = %w[
+        type_of_gt12
+        send_application_alerts
+        application_alert_email
+        gt12_response_destination
+      ]
+      expect(new_provider.ucas_preferences.attributes.slice(compare_attrs))
+        .to eq provider.ucas_preferences.attributes.slice(compare_attrs)
+    end
+
+    it 'copies over the contacts' do
+      service.execute(new_recruitment_cycle)
+
+      compare_attrs = %w[name email telephone]
+      expect(new_provider.contacts.map { |c| c.attributes.slice(compare_attrs) })
+        .to eq(provider.contacts.map { |c| c.attributes.slice(compare_attrs) })
+    end
+
+    it 'copies over the sites' do
+      service.execute(new_recruitment_cycle)
+
+      expect(mocked_copy_site_service).to have_received(:execute).with(site: site, new_provider: new_provider)
+    end
+
+    it 'copies over the courses' do
+      service.execute(new_recruitment_cycle)
+
+      expect(mocked_copy_course_service).to have_received(:execute).with(course: course, new_provider: new_provider)
+    end
+
+    it 'returns a hash of the counts of copied objects' do
```

Comment:
Should also test the value it returns also :eyes: you can mock that value for the sites and courses in your doubles, ensures that they're wired together as expected